compiler/main/driver.cpp

```diff
@@ -1502,14 +1504,26 @@ static void setPrintCppLineno() {
   if (developer && !userSetCppLineno) printCppLineno = false;
 }
 
-static void setSaveCForGpuCodegen() {
+static void setGPUFlags() {
   bool isGpuCodegen = localeUsesGPU();
   bool isSaveCDirEmpty = strlen(saveCDir) == 0;
 
-  if(isGpuCodegen && isSaveCDirEmpty) {
-    int len = snprintf(saveCDir, FILENAME_MAX+1, "%s_gpu_files", executableFilename);
-    if(len < 0 || len >= FILENAME_MAX+1) {
-      USR_FATAL("Unable to produce name of savec directory for GPU code generation.");
+  if(isGpuCodegen) {
+    if (isSaveCDirEmpty) {
+      const char *gpuDir = strlen(executableFilename) != 0 ? 
+            executableFilename:std::to_string(getpid()).c_str();
```

Comment:
In what situation is executableFilename the empty string?

Reply:
This variable is only set if there's `-o` in the compilation line. Without this particular change, we were creating dirs named `_gpu_files` if the user doesn't use `-o`. Note that we use `-o` in start_test.

It looks like we don't create the executable name until codegen, if there's no `-o`. I think we need resolution (or scope resolution?) to have happened before we can determine the executable name. Arguably, things could be rewired around all this, but I didn't want to spend the effort for something temporary.